File: kits/java/simple/lux/Player.java
package lux;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class Player {
  public int team;
  public int researchPoints;
  public ArrayList<Unit> units = new ArrayList<>();
  public Map<String, City> cities = new HashMap<>();
  public int cityTileCount;
  public Player(int team) {
    this.team = team;
    this.researchPoints = 0;
    this.cityTileCount = 0;
  }
  public boolean researchedCoal() {
    return this.researchPoints >= GameConstants.PARAMETERS.RESEARCH_REQUIREMENTS.COAL;
  }
  public boolean researchedUranium() {
    return this.researchPoints >= GameConstants.PARAMETERS.RESEARCH_REQUIREMENTS.URANIUM;
  }
}

File: kits/ts/simple/lux/Player.ts
import GAME_CONSTANTS from "./game_constants.json";
import {City} from "./City";
import {Unit} from "./Unit";

/**
 * holds all data related to a player
 */
export class Player {
  public readonly team: number;
  public researchPoints = 0;
  // Map unit id to the unit
  public units = new Array<Unit>();
  public cities = new Map<string, City>();
  public cityTileCount = 0;

  public constructor(teamid: number) {
    this.team = teamid;
  }

  public researchedCoal(): boolean {
    return this.researchPoints >= GAME_CONSTANTS.PARAMETERS.RESEARCH_REQUIREMENTS.COAL;
  }

  public researchedUranium(): boolean {
    return this.researchPoints >= GAME_CONSTANTS.PARAMETERS.RESEARCH_REQUIREMENTS.URANIUM;
  }
}